src/QueryEditor/QueryEditor.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { css } from '@emotion/css';
+import { QueryEditorProps } from '@grafana/data';
+import { CentreonDataSource } from '../centreonDataSource';
+import { CentreonMetricOptions, defaultQuery, MyQuery } from '../@types/types';
+import { debounce, defaults } from 'lodash';
+import { QueryEditorModeSwitcher } from './QueryEditorModeSwitcher';
+import { EMode } from './EMode';
+import { VisualCentreonEditor } from './VisualCentreonEditor';
+import { RawCentreonQueryEditor } from './RawCentreonQueryEditor';
+
+type Props = QueryEditorProps<CentreonDataSource, MyQuery, CentreonMetricOptions>;
+
+export const QueryEditor: React.FC<Props> = (props: Props) => {
+  const { query, onRunQuery, onChange: pOnchange, datasource } = props;
```

Comment:
pOnChange doesn't sound clear to me, generally speaking, we can use the 'onXXX' notation for callback, and name the functions to use a verb. You can change the function below to name it 'change' or even 'handleChange'